File: backend/internal/models/filiere.go
package models

import "time"

type Filiere struct {
	ID        int       `json:"id" db:"id"`
	Code      string    `json:"code" db:"code"`
	Label     string    `json:"label" db:"label"`
	Color     string    `json:"color" db:"color"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
}

type FilieresData struct {
	Filieres []Filiere `json:"filieres"`
}

File: frontend/src/types/filiere.ts
export interface Filiere {
    id: number
    code: string
    label: string
    color: string
}

export interface FilieresResponse {
    filieres: Filiere[]
}

export interface CreateFiliereData {
    code: string
    label: string
    color: string
}

export interface UpdateFiliereData {
    code?: string
    label?: string
    color?: string
}
